src/math/peaks.cpp

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2024 Team Dissolve and contributors
+
+#include "math/peaks.h"
+#include "math/mathFunc.h"
+#include <algorithm>
+
+/*
+ * Peaks
+ */
+Peaks::Peaks(const std::vector<double> &values, const std::vector<double> &domain) : values_(values), domain_(domain)
+{
+    maxValue_ = *std::max_element(values_.begin(), values_.end());
+    resetConstraints();
+}
+
+Peaks::Peaks(const Data1D &source) : values_(source.values()), domain_(source.xAxis())
+{
+    maxValue_ = *std::max_element(values_.begin(), values_.end());
+    resetConstraints();
+}
+
+// Set vertical threshold for peaks
+void Peaks::setThreshold(double threshold) { threshold_ = threshold; }
+
+// Set horizontal isolation radius for peaks
+void Peaks::setIsolation(double radius) { isolation_ = radius; }
+
+// Reset the threshold and isolation constraints
+void Peaks::resetConstraints() { threshold_ = -10e9, isolation_ = 0; }
+
+// Check if neighbouring points correspond to a local maximum
+bool Peaks::isLocalMaximum(double *points) { return (points[1] - points[0] > 0) && (points[2] - points[1] <= 0); }
+
+// Check if neighbouring points correspond to a local minimum
+bool Peaks::isLocalMinimum(double *points)
+{
+    double invertedPoints[3] = {-1.0 * points[0], -1.0 * points[1], -1.0 * points[2]};
+    return isLocalMaximum(invertedPoints);
+}
+
+// Check if neighbouring points correspond to an inflection point
+bool Peaks::isInflectionPoint(double *points)
+{
+    return (((points[1] - points[0] > 0) || ((points[1] - points[0] < 0))) && (points[2] - points[1] == 0)) ||
+           (((points[2] - points[1] > 0) || ((points[2] - points[1] < 0))) && (points[1] - points[0] == 0));
+}
+
+// Sort peaks in place by peak value, from highest to lowest
+void Peaks::sortPeaks(std::vector<Peaks::Peak1D> &peaks)
+{
+    std::sort(peaks.begin(), peaks.end(), [](const auto &a, const auto &b) { return a.peak > b.peak; });
+}
+
+// Sort peaks in place by index, from first to last
+template <typename T> void Peaks::sortIndices(std::vector<T> &items)
+{
+    std::sort(items.begin(), items.end(), [](const auto &a, const auto &b) { return a.index < b.index; });
+}
+
+// Sort prominences in place by prominence value, from highest to lowest
+void Peaks::sortProminences(std::vector<Peaks::Prominence1D> &proms)
+{
+    std::sort(proms.begin(), proms.end(), [](const auto &a, const auto &b) { return a.prominence > b.prominence; });
+}
+
+// Get top n peaks
+std::vector<Peaks::Peak1D> Peaks::top(int n, std::vector<Peaks::Peak1D> peaks)
```

Comment:
```suggestion
std::vector<Peaks::Peak1D> Peaks::top(std::size_t n, std::vector<Peaks::Peak1D> peaks)
```

This is pedantic, but `n` must logically be between 0 and the largest possible size of `peaks`, which is the exact definition of `std::size_t`.  If also saves us and `abs` call later on.